### PR TITLE
compat: update compatibility matrix for 4.20/main onwards

### PR DESCRIPTION
**Pull request by rohityadavcloud**
Related PR: 
https://github.com/apache/cloudstack/pull/8609

----
📚 Documentation preview 📚: https://cloudstack-documentation--409.org.readthedocs.build/en/409/

### CODE REVIEW

**Review by sureshanaparti — APPROVED** [review]
LGTM

**Review by JoaoJandre — APPROVED** [review]
LGTM

File: source/releasenotes/compat.rst [hunk 1]
```diff
@@ -23,55 +23,51 @@ This section lists the operating systems that are supported for running
...
 
-   .. note:: There is a known issue with ipmitool with RHEL8 / SUSE, so certain functionality such as out of band management might not work
+   .. note:: There is a known issue with ipmitool with EL8 / EL9 / SUSE, so certain functionality such as out of band management might not work
 
 Software Requirements
 ~~~~~~~~~~~~~~~~~~~~~
 
--  Java JRE 17
--  MySQL 5.6, 5.7, 8.0
+-  Java JRE 17, Java JRE 21
```
[the remark this answers is not in the document]
Comment (weizhouapache) [reply source/releasenotes/compat.rst:39]: Yes @rohityadavcloud 
The packages for PRs targeted main/4.20.0.0-SNAPSHOT is actually packaged using JDK11, but mgmt/Agent/systemvms are using JRE17 which look good, no compatibility issue found. I am not sure if it works with JRE21.

Let's discuss on mailing list

Comment (rohityadavcloud) [inline source/releasenotes/compat.rst:39]:
```suggestion
-  Java JRE 11 or Java JRE 17

File: source/releasenotes/compat.rst [hunk 2]
```diff
@@ -23,55 +23,51 @@ This section lists the operating systems that are supported for running
...
-   .. note:: There is a known issue with ipmitool with RHEL8 / SUSE, so certain functionality such as out of band management might not work
+   .. note:: There is a known issue with ipmitool with EL8 / EL9 / SUSE, so certain functionality such as out of band management might not work
 
 Software Requirements
 ~~~~~~~~~~~~~~~~~~~~~
 
--  Java JRE 17
--  MySQL 5.6, 5.7, 8.0
+-  Java JRE 11 or Java JRE 17
+-  MySQL 8.0
```
Comment (rohityadavcloud) [inline source/releasenotes/compat.rst:40]:
```suggestion
-  MySQL 8.0 (or equivalent compatible DBMS)
```

File: source/releasenotes/compat.rst [hunk 3]
```diff
@@ -23,55 +23,51 @@ This section lists the operating systems that are supported for running
...
 
-   .. note:: There is a known issue with ipmitool with RHEL8 / SUSE, so certain functionality such as out of band management might not work
+   .. note:: There is a known issue with ipmitool with EL8 / EL9 / SUSE, so certain functionality such as out of band management might not work
 
 Software Requirements
 ~~~~~~~~~~~~~~~~~~~~~
 
--  Java JRE 17
--  MySQL 5.6, 5.7, 8.0
+-  Java JRE 11 or Java JRE 17
```
Comment (weizhouapache) [inline source/releasenotes/compat.rst:39]:
```suggestion
-  Java JRE 17
```

for main/4.20, the java/jre version must be 17 if OS is centos8/9 or ubuntu

ubuntu
https://github.com/apache/cloudstack/blob/b69cc0272d48f0aea7353627d760c27c284dad84/debian/control#L20
https://github.com/apache/cloudstack/blob/b69cc0272d48f0aea7353627d760c27c284dad84/debian/control#L27

centos8/9:
https://github.com/apache/cloudstack/blob/b69cc0272d48f0aea7353627d760c27c284dad84/packaging/centos8/cloud.spec#L55
https://github.com/apache/cloudstack/blob/b69cc0272d48f0aea7353627d760c27c284dad84/packaging/centos8/cloud.spec#L101

Comment (rohityadavcloud) [reply source/releasenotes/compat.rst:39]: As this is merged now, feel free to raise a new PR to correct this

**Pull request merged**